### ISSUE TITLE
Translations

### ISSUE THREAD

**Issue by Zireael07** [gabordemooij/citrine#139]
Citrine claims to be multilingual and the page has a subpage about how to provide a translation. However what I see in the docs looks like machine translation (leading to hilariously bad results for inflectional languages like most Slavic ones)

**Comment by gabordemooij** [comment]
One has to start somewhere, right? For several languages, like Russian, Polish, and Uzbek, we’re fortunate to have community members who help with translations. For other languages, we currently rely on machine translation. There are two main reasons for this. First, it gives people a sense of how Citrine could look in their language. Second, it encourages native speakers to step in and improve any inaccuracies from the machine translations—which has led to many valuable contributions so far!

I'm always transparent about using machine translation. What’s your native language?

**Comment by Zireael07** [comment]
Polish. 

https://citrine-lang.org/docs/pl/ac7c9d77c4fa36f4f8cb782f5d71ef275d467a1e.html at least looks like partly machine translated (Obiekt | robić, Obiekt | siebie, Strunowy | haszysz )... Russian and Czech seem to suffer from the same problem and the page doesn't have any disclaimer about MT being used

No clue how Text became strunowy? I could understand it if the English original were String but again, it is a hallmark of machine translation.... String in programming is usually translated as tekst (text), struna is a string but in a musical instrument.
Haszysz is a drug/narc called hash in English. In programming context, it is usually "haszowanie" (verb) or "funkcja skrótu"

**Comment by gabordemooij** [comment]
Looking at the credits page (https://citrine-lang.org/credits.ctr), it seems like Jan worked on these translations. It’s a bit puzzling, and I don’t really have an explanation for it. As you can imagine, I’m unable to personally verify every translation, though I do try sometimes by checking files with back-and-forth translations. Roman is our Russian translator, and while he’s not on GitHub, I can easily reach out to him - we're in good contact.

Actually, adding a machine translation (MT) disclaimer is a great idea! I’ll work on that, though it might take some time since there’s quite a pipeline behind the manual.

In the meantime, if you have any suggestions or improvements for the Polish version, feel free to send them my way or open a pull request (PR). All contributions are greatly appreciated!

**Comment by gabordemooij** [comment]
I also downgraded the progress of Polish and Russian to 60% to reflect this degraded quality. (https://citrine-lang.org/add.ctr).

**Comment by Zireael07** [comment]
> t’s a bit puzzling, and I don’t really have an explanation for it. 

I haven't looked into your translation files (yet - do they specify the context? I know some translation formats allow for a context specifier(s)) but it is possible the person just pulled the first meaning from a paper (or online) dictionary without checking the context at all. It is possible the person is NOT a programmer at all so wouldn't have a clue what a String or hash are, too.

> As you can imagine, I’m unable to personally verify every translation,

That is why I am majorly skeptical of projects like this, that claim to be translated into everything under the sun. There is no way the author can verify the translations and unless you are very certain of the translators you picked, you can end up with a mess anyway :(

**Comment by gabordemooij** [comment]
Yes, the translation templates are annotated and specify context - I also included them in the repo now: https://github.com/gabordemooij/citrine/blob/master/misc/translator/dictionary.h.

> That is why I am majorly skeptical of projects like this, that claim to be translated into everything under the sun. There is no way the author can verify the translations and unless you are very certain of the translators you picked, you can end up with a mess anyway :(

The concept behind the project is quite simple: I want to code in my own language, and if others feel the same, they’re welcome to join. While I understand some may have concerns about the language files being a bit messy, I believe that if someone truly cares about a particular language, they’ll contribute by improving those files. If not, that’s okay too.

When I rebooted this project in 2009 (the original idea dates back to 1993, starting with an implementation on the Commodore 64), I hoped that people would step in and create language files themselves. Although a few people liked the idea, nothing really came of it. So instead of waiting around for things to happen, I decided to generate all the language files myself using machine translation and make them available to the public. That’s how the project came to life.

Some files may stay in a rough state until a native speaker has the time and interest to enhance them - and that's part of the project’s design. It’s meant to evolve through community contributions when the time is right.
